lib_runtime/functoria/functoria_runtime.ml

```diff
@@ -25,9 +26,14 @@ module Arg = struct
   let get t =
     match t.value with
     | None ->
-        invalid_arg
-          "Functoria_runtime.Arg..get: Called too early. Please delay this \
-           call after cmdliner's evaluation."
+        if !initialized then
+          invalid_arg
+            "Not sure how you got here, please report an issue at the issue \
+             tracker with your unikernel source."
```

Comment:
Ah! I guess it can happen with
```OCaml
let hello =
    let doc = Arg.info ~doc:"How to say hello." [ "hello" ] in
    Mirage_runtime.register Arg.(value & opt string "Hello World!" doc)
let hello = hello ()
module Make(_ : module type of Unit) = struct
  let start () =
    Logs.app (fun m -> m "%s" hello); Lwt.return_unit
end
```
That is, `Mirage_runtime.register` is called in a timely manner, but `hello ()` is called too early.

Reply:
Yes, but then only the `else` branch is hit -- where `initialized` is false. I couldn't wrap my head around an example where the `if !initalized` is hit -- so maybe the entire conditional in `get` can be removed, and we've the error in `register` and the other in `get`!?